src/Tickets/Order_Modifiers/Modifiers/Coupon.php

```
@@ -0,0 +1,150 @@
+
+	/**
+	 * The modifier type for coupons.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $modifier_type = 'coupon';
+
+	/**
+	 * Gets the modifier type for coupons.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The modifier type ('coupon').
+	 */
+	public function get_modifier_type(): string {
+		return $this->modifier_type;
+	}
+
+	/**
+	 * Inserts a new Coupon Modifier.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data The data to insert.
+	 *
+	 * @return mixed The newly inserted modifier or an empty array if no changes were made.
+	 */
+	public function insert_modifier( array $data ): mixed {
+		// Ensure the modifier_type is set to 'coupon'.
+		$data['modifier_type'] = $this->modifier_type;
+
+		// Validate data before proceeding.
+		if ( ! $this->validate_data( $data ) ) {
+			return [];
+		}
+
+		// Use the repository to insert the data into the `order_modifiers` table.
+		$repository = new Order_Modifiers_Repository();
+		return $repository->insert( new Order_Modifier( $data ) );
+	}
+
+	/**
+	 * Updates an existing Coupon Modifier.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data The data to update.
+	 *
+	 * @return mixed The updated modifier or an empty array if no changes were made.
+	 */
+	public function update_modifier( array $data ): mixed {
+		// Ensure the modifier_type is set to 'coupon'.
+		$data['modifier_type'] = $this->modifier_type;
+
+		// Validate data before proceeding.
+		if ( ! $this->validate_data( $data ) ) {
+			return [];
+		}
+
+		// Use the repository to update the data in the `order_modifiers` table.
+		$repository = new Order_Modifiers_Repository();
+		return $repository->update( new Order_Modifier( $data ) );
+	}
+
+	/**
+	 * Validates the required fields for Coupons.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data The data to validate.
+	 *
+	 * @return bool True if the data is valid, false otherwise.
+	 */
+	public function validate_data( array $data ): bool {
```

Rather than returning `true` or `false`, it should silently pass, or throw an exception for failing.